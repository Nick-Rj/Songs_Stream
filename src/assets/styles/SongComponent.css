.SongComponent {
  color: black;
}

.text {
  color: black !important;
}

.right {
  position: absolute;
  right: 0;
  float: right;
  margin-right: 5em;
}

.inline-component {
  display: inline-block !important;
}

.Popover-styling {
  width: 40em !important;
}
