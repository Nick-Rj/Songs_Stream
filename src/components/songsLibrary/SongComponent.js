import React, { useState } from "react";
import { Button, Card, Modal } from "react-bootstrap";
import { withRouter, useHistory } from "react-router-dom";
import { deleteSong } from "../../redux/store";
import { connect } from "react-redux";

import "../../assets/styles/SongComponent.css";
const SongComponent = withRouter(
  ({ deleteSong, userLoggedInStatus, song, isSongsLibrary }) => {
    const toUpdateSong = useHistory();
    const handleUpdate = (data) => {
      toUpdateSong.push({
        pathname: "/updateSong",
        state: data,
      });
    };

    const [show, setShow] = useState(false);

    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);

    const handleDeleteSong = () => {
      alert(`${song.title} delete successfully!`);
      deleteSong(song.id);

      handleClose();
    };

    const urlValidator = (event) => {
      if (song.url === "") {
        alert("Song not found! Please provide a proper url...");
        event.preventDefault();
      }
    };

    return (
      <div className="SongComponent">
        <Card style={{ width: "100%" }}>
          <Card.Body>
            <Card.Title>{song.title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">
              {song.singer}
            </Card.Subtitle>
            <br />
            <Card.Text className="text">
              <span>
                <h6>Album/Movie:</h6>
              </span>
              <span>{song.album_movie}</span>
            </Card.Text>
            <Card.Text className="text">
              <span>
                <h6>Genre:</h6>
              </span>
              <span>{song.genre}</span>
            </Card.Text>
            <Card.Text className="text">
              <span>
                <h6>Song Length:</h6>
              </span>
              <span>{song.length}</span>
            </Card.Text>
            <Card.Link href={song.url} target="_blank">
              <Button
                variant="primary"
                className="Button"
                onClick={urlValidator}
              >
                Listen on Youtube
              </Button>
            </Card.Link>
            {userLoggedInStatus && isSongsLibrary && (
              <div className="inline-component">
                <Button
                  variant="success"
                  className="Button"
                  onClick={() => handleUpdate(song)}
                >
                  Update
                </Button>
              </div>
            )}

            {userLoggedInStatus && isSongsLibrary && (
              <div className="inline-component">
                <Button variant="danger" onClick={handleShow}>
                  Delete
                </Button>

                <Modal
                  show={show}
                  onHide={handleClose}
                  backdrop="static"
                  keyboard={false}
                >
                  <Modal.Header closeButton>
                    <Modal.Title>Delete Song</Modal.Title>
                  </Modal.Header>
                  <Modal.Body>
                    {song.title} will be deleted permanently. Are you sure?
                  </Modal.Body>
                  <Modal.Footer>
                    <Button variant="secondary" onClick={handleClose}>
                      Cancel
                    </Button>
                    <Button variant="danger" onClick={handleDeleteSong}>
                      Delete
                    </Button>
                  </Modal.Footer>
                </Modal>
              </div>
            )}
          </Card.Body>
        </Card>
      </div>
    );
  }
);

export default connect(null, { deleteSong })(SongComponent);
