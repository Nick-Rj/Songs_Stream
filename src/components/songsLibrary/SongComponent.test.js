import { shallow } from "enzyme";
import { expect, it } from "@jest/globals";

import SongComponent from "./SongComponent";

import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({
  userState: {
    loading: false,
    users: [],
    user: {},
    error: "",
    isLoggedIn: false,
  },
  songData: { loading: false, songs: [], error: "" },
  playlistData: { loading: false, playlists: [], error: "" },
});

it("rendering full DOM for SongComponent Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <SongComponent />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
