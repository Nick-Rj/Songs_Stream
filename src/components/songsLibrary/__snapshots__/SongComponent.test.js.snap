// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering full DOM for SongComponent Component 1`] = `ShallowWrapper {}`;
